File: Singleton.php
<?php 
final class Pointer{
    private static ?Pointer $pointer = null; //Private static atributte
    private function __construct(){} //private constructor
    public static function createSingleton(): Pointer{
        if( self::$pointer === null ){
            self::$pointer = new Pointer();
        }else{
            echo ' Object already created';
        }
        return self::$pointer;
    }
}
    $p = Pointer::createSingleton();
    $p1 = Pointer::createSingleton();
    
?>

File: Singleton.cpp
#include<iostream>
//Declaration class
class Singleton{
    private: 
    static Singleton *singleton;
    Singleton(){
        std::cout<<"Object created"<<std::endl;
    }
    public: 
    static Singleton* createSingleton();

};
//Definition class
Singleton *Singleton::singleton = 0;
Singleton *Singleton::createSingleton(){
    if (!singleton)
    {
        singleton = new Singleton();
        return singleton;
    }
    else
    {
        std::cout << "Already created" << std::endl;
    }
    return singleton;
}

int main(int argc, char const *argv[]){
    /* code */
    Singleton *singleton1 = Singleton::createSingleton();
    Singleton *singleton2 = Singleton::createSingleton();
    return 0;
}
